homeassistant/components/vera/lock.py

```diff
@@ -44,6 +47,48 @@ def is_locked(self):
         """Return true if device is on."""
         return self._state == STATE_LOCKED
 
+    @property
+    def device_state_attributes(self):
+        """Who unlocked the lock and did a low battery alert fire.
+
+        Reports on the previous poll cycle.
+        changed_by_name is a string like 'Bob'.
+        low_battery is 1 if an alert fired, 0 otherwise.
+        """
+        data = super().device_state_attributes
+
+        # This check will no longer be required once the version of the pyvera
+        # library with these two functions included is generally available.
+        if not ("get_last_user_alert" in dir(self.vera_device)) and (
+            "get_low_battery_alert" in dir(self.vera_device)
+        ):
+            return data
+
+        last_user = self.vera_device.get_last_user_alert()
+        if last_user is not None:
+            data[ATTR_LAST_USER_NAME] = last_user[1]
+
+        data[ATTR_LOW_BATTERY] = self.vera_device.get_low_battery_alert()
+        return data
+
+    @property
+    def changed_by(self):
+        """Who unlocked the lock.
+
+        Reports on the previous poll cycle.
+        changed_by is an integer user ID.
+        """
+        # This check will no longer be required once the version of the pyvera
+        # library with these two functions included is generally available.
```

Comment:
We should wait for the pyvera bump before we add this.

Reply:
Yeah, I didn't know how you typically handled libraries and HomeAssistant being on different release cycles.  So I just included code to make sure it would not fail no matter what order a user upgraded their code in.

Reply:
If pyvera is released with this feature it's ok to bump the pyvera version that home assistant uses in this PR. Until then we have to put this PR on hold.

Reply:
Makes sense to me.